### PR TITLE
MES  1404 - Airwatch Spike

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

This PR allows us to test that we can get app config from air watch

**DO NOT MERGE - USE AS A GUIDE FOR THE FINAL VERSION**

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [ ] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -22,6 +23,7 @@ export class App {
       .then(() => {
         this.configureStatusBar();
         this.loadAppInfo();
+        this.loadAirwatchConfig();
```
Comment (ammarhaiderbjss) [inline src/app/app.component.ts]: Could this cause a race condition? We need this config endpoint immediately after log in, maybe it should be in `login.ts`?

File: airwatch-config/config.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ammarhaiderbjss) [inline airwatch-config/config.xml]: Just realised we also need the Azure AD constants in here too for the app to be able to log in, so all this -
```
  authentication: {
    context: 'https://login.windows.net/common',
    resourceUrl: '09fdd68c-4f2f-45c2-be55-dd98104d4f74',
    clientId: '09fdd68c-4f2f-45c2-be55-dd98104d4f74',
    redirectUrl: 'x-msauth-uk-gov-dvsa-mobile-examiner://uk.gov.dvsa.mobile-examiner',
    logoutUrl: 'https://login.windows.net/6c448d90-4ca1-4caf-ab59-0a2aa67d7801/oauth2/logout',
    employeeIdKey: 'extn.employeeId',
  },
```

Comment (Cavalier07) [reply airwatch-config/config.xml]: Have updated the config to provide the values , have also added a new selector to pull the auth config from the store.

File: src/app/app.component.ts (hunk 1 above)
Comment (Cavalier07) [reply src/app/app.component.ts]: Have moved it to login where we dispatch the action and subscribe to the results passing them into the login function

File: package.json [hunk 3]
```diff
@@ -63,15 +64,16 @@
     "@ngrx/store": "^5.2.0",
     "cordova-ios": "4.5.5",
     "cordova-plugin-app-version": "0.1.9",
-    "cordova-plugin-compat": "1.2.0",
+    "cordova-plugin-app-preferences": "0.99.3",
+    "cordova-plugin-compat": "^1.2.0",
     "cordova-plugin-device": "^2.0.2",
-    "cordova-plugin-google-analytics": "1.8.6",
-    "cordova-plugin-inappbrowser": "3.0.0",
+    "cordova-plugin-google-analytics": "^1.8.6",
```
Comment (ammarhaiderbjss) [inline package.json]: can we revert these changes to versions?

File: src/providers/airwatch-config/airwatch-config.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor(private appPreferences: AppPreferences) {}
+
+  public getAirwatchConfig = (): Observable<AirwatchConfigModel> =>  {
+    return fromPromise(this.getConfig());
+  }
+
+  private getConfig = async(): Promise<AirwatchConfigModel> => {
+    return {
+      configUrl: await this.appPreferences.fetch('configUrl'),
```
Comment (ammarhaiderbjss) [inline src/providers/airwatch-config/airwatch-config.ts]: Assuming there's no fetch all?

Comment (Cavalier07) [reply src/providers/airwatch-config/airwatch-config.ts]: No there's a defaults option which will get all the default values but that restricts how we can use the config xml.

File: package.json (hunk 3 above)
Comment (Cavalier07) [reply package.json]: Done
